src/print.c

```diff
@@ -245,6 +255,190 @@ launch_preview (const char *filename,
   return retval;
 }
 
+static int
+pdf_get_actions_page (char        *filename,
+                      PDF_ACTIONS  act,
+                      int          page)
+{
+  GSubprocess *process;
+  GInputStream *stream;
+  char *option1 = NULL;
+  char *option2 = NULL;
+  char buffer[50] = { 0 };
+
+  option1 = g_strconcat ("--file=", filename, NULL);
+  switch(act)
+  {
+    case PDF_TEST :
+       (void)page;
+       option2 = g_strdup ("--test");
+       break;
+    case PDF_NUM_PAGES :
+       (void)page;
+       option2 = g_strdup ("--pages");
+       break;
+    case PDF_WIDTH :
+       option2 = g_strdup_printf ("--width=%d", page);
+       break;
+    case PDF_HEIGHT :
+       option2 = g_strdup_printf ("--height=%d", page);
+       break;
+   default:
+       return 0;
+  }
+  process = g_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE, NULL, "/usr/libexec/pdftoraw", option1, option2, NULL);
+
+  if (process)
+  {
+    stream = g_subprocess_get_stdout_pipe(process);
+    if (stream)
+    {
+      g_input_stream_read(stream, buffer, sizeof(buffer), NULL, NULL);
+      if (buffer[0])
+        return atoi(buffer);
+    }
+  }
+  return 0;
+}
+
+static unsigned char *
+pdf_get_data_page (char *filename,
+                   int   w,
+                   int   h,
+                   int   page)
+{
+  GSubprocess *process;
+  GInputStream *stream;
+  unsigned char *data = NULL;
+  char *option1 = NULL;
+  char *option2 = NULL;
+  int read = 0;
+  gsize total_read = 0;
+
+  option1 = g_strconcat ("--file=", filename, NULL);
+  option2 = g_strdup_printf ("--raw=%d", page);
+  process = g_subprocess_new(G_SUBPROCESS_FLAGS_STDOUT_PIPE, NULL, "/usr/libexec/pdftoraw", option1, option2, NULL);
+
+  if (process)
+  {
+    stream = g_subprocess_get_stdout_pipe(process);
+    if (stream)
+    {
+      data = (unsigned char *) g_malloc( w * h * 4);
+      while ((read = g_input_stream_read(stream, data + total_read, 1024, NULL, NULL)) > 0)
+      {
+         total_read += read;
+      }
+    }
+  }
+  return data;
+}
+
+static void
+pdf_begin_print(GtkPrintOperation *op,
+                GtkPrintContext   *ctx,
+                char              *filename)
+{
+    (void)ctx;
```

Comment:
I don't think this will warn with our flags. Use [`G_GNUC_UNUSED`](https://developer-old.gnome.org/glib/stable/glib-Miscellaneous-Macros.html#G-GNUC-UNUSED:CAPS) above if it does.

Reply:
Fixed